openmc/openmoc_compatible.py

```diff
@@ -0,0 +1,747 @@
+    # works if the region is a single half-space or an intersection of
+    # half-spaces, i.e., no complex cells.
+    region = openmc_cell.region
+    if region is not None:
+        if isinstance(region, openmc.Halfspace):
+            surface = region.surface
+            halfspace = -1 if region.side == '-' else 1
+            openmoc_cell.addSurface(halfspace, get_openmoc_surface(surface))
+        elif isinstance(region, openmc.Intersection):
+            for node in region.nodes:
+                if not isinstance(node, openmc.Halfspace):
+                    raise NotImplementedError("Complex cells not yet "
+                                              "supported in OpenMOC.")
+                surface = node.surface
+                halfspace = -1 if node.side == '-' else 1
+                openmoc_cell.addSurface(halfspace, get_openmoc_surface(surface))
+        else:
+            raise NotImplementedError("Complex cells not yet supported "
+                                      "in OpenMOC.")
+
+    # Add the OpenMC Cell to the global collection of all OpenMC Cells
+    OPENMC_CELLS[cell_id] = openmc_cell
+
+    # Add the OpenMOC Cell to the global collection of all OpenMOC Cells
+    OPENMOC_CELLS[cell_id] = openmoc_cell
+
+    return openmoc_cell
+
+
+def get_openmc_cell(openmoc_cell):
+    """Return an OpenMC cell corresponding to an OpenMOC cell.
+
+    Parameters
+    ----------
+    openmoc_cell : openmoc.Cell
+        OpenCG cell
+
+    Returns
+    -------
+    openmc_cell : openmc.universe.Cell
+        Equivalent OpenMC cell
+
+    """
+
+    cv.check_type('openmoc_cell', openmoc_cell, openmoc.Cell)
+
+    cell_id = openmoc_cell.getId()
+
+    # If this Cell was already created, use it
+    if cell_id in OPENMC_CELLS:
+        return OPENMC_CELLS[cell_id]
+
+    # Create an OpenMOC Cell to represent this OpenMC Cell
+    name = openmoc_cell.getName()
+    openmc_cell = openmc.Cell(cell_id, name)
+
+    if (openmoc_cell.getType() == openmoc.MATERIAL):
+        fill = openmoc_cell.getFillMaterial()
+        openmc_cell.fill = get_openmc_material(fill)
+    elif (openmoc_cell.getType() == openmoc.FILL):
+        fill = openmoc_cell.getFillUniverse()
+        if isinstance(fill, openmoc.Lattice):
+            openmc_cell.fill = get_openmc_lattice(fill)
+        else:
+            openmc_cell.fill = get_openmc_universe(fill)
+
+    if openmoc_cell.isRotated():
+        rotation = openmoc_cell.getRotation(3)
+        openmc_cell.rotation = rotation
+    if openmoc_cell.isTranslated():
+        translation = openmoc_cell.getTranslation(3)
+        openmc_cell.translation = translation
+
+
+    surfaces = []
+    operators = []
+    for surf_id, surf_halfspace in openmoc_cell.getSurfaces().values():
+        halfspace = surf_halfspace._halfspace
+        surface = surf_halfspace._surface
+        surfaces.append(get_openmc_surface(surface))
+        operators.append(operator.neg if halfspace == -1 else operator.pos)
```

Comment:
Maybe `regions = []` above and then `regions.append(-surface if halfspace == -1 else +surface)` and `openmc_cell.region = openmc.Intersection(*regions)`